### PR TITLE
Add is_integral method to algebraic numbers

### PR DESCRIPTION
**Pull request by user202729**
For symmetry with existing methods e.g. `QQ(1).is_integral` or such method on elements of number fields.

Implementation pretty much copied from `NumberFieldElement`.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/rings/qqbar.py [hunk 1]
```diff
@@ -4485,6 +4485,26 @@ def as_number_field_element(self, minimal=False, embedded=False, prec=53):
         """
         return number_field_elements_from_algebraics(self, minimal=minimal, embedded=embedded, prec=prec)
 
+    def is_integral(self):
+        r"""
+        Determine if a number is an algebraic integer.
+
```
Comment (kwankyu) [inline src/sage/rings/qqbar.py:4491]: This description does not read well. The method works with the given object, namely, ``self``.  It is not just "a number". I suggest
```
Check if this number is an algebraic integer.
```
or
```
Return ``True`` if this number is an algebraic integer.
```

Comment (user202729) [reply src/sage/rings/qqbar.py:4491]: Okay. 

(Result of mindlessly copying from the other method I guess. In `number_field_element.pyx`:

```python
    def is_integral(self):
        r"""
        Determine if a number is in the ring of integers of this number
        field.

```
)

Comment (user202729) [reply src/sage/rings/qqbar.py:4491]: Fixed, thanks for the review.

**Review by kwankyu — APPROVED** [review]
Thanks. LGTM.

**Pull request merged**